nginx/detect.go

```diff
@@ -0,0 +1,54 @@
+package nginx
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/cloudfoundry/packit"
+)
+
+//go:generate faux --interface VersionParser --output fakes/version_parser.go
+type VersionParser interface {
+	ParseVersion(workingDir, cnbPath string) (version, versionSource string, err error)
+}
+
+type BuildPlanMetadata struct {
+	VersionSource string `toml:"version-source,omitempty"`
+}
+
+func Detect(versionParser VersionParser) packit.DetectFunc {
```

Comment:
Is it possible to have a case where a `buildpack.yml` file contains a version of nginx, but there is no `nginx.conf`? Specifically, I am thinking of a case for PHP apps that might have this scenario. Does that make sense, or would this scenario just never happen?

Reply:
The `php-web` buildpack relies on this `detect` behavior. They generate a `nginx.conf`

Reply:
Yes it is. The nginx cnb in that case provides the dependency. The lifecycle-detector will fail if that dependency isn't asked for by another cnb.
Added unit and integration test cases where there's an app with no nginx conf. Expected behavior being nginx-cnb detect passes but the build will fail since "nginx" is unused.